Site react

EllarBooher opened the pull request:
I'm not confident this is ready to be merged, but I got both production and development builds running locally, and I'd like to get feedback and discuss some of the changes. Here's a summary of my work:
- I did my best to preserve the existing folder structure, while integrating Vite. All components are in `resources/components`, build is still output to `public.build`, etc.
- build.js is gone, Vite + a plugin handles all of that in `vite.config.mts`
- For development with hot reloading, start Vite first with `npm run dev`
- I added the npm build step to the dockerfile

Here are some issues with I am aware of that I was still working on, which I can finish up before merging if you'd like:
- The frontend is not responsive to thin mobile formats and just breaks
- Inconsistent documentation and a few very bloated files
- There is no hovered tile indicator on map
- There is no demo page / mock API

I also brought in my lint and format preferences, but I don't want to unilaterally decide that. I'm using eslint with some rules set up in `eslint.config.mjs`. For formatting, there's prettier for Javascript and stylelint for CSS.

wouterrutgers commented:
Awesome! I haven't used React before, so while I might not catch every detail, I do have some thoughts to share (I am fluent in Vue, similar concepts, just different syntax, right).

If you think an different directory structure makes things easier that would be fine by me!

Having a basic overview on mobile would be nice to have, so that's something we should get working again.

For testing, I've just been connecting directly to the live endpoint, but I agree that having a dedicated test endpoint with mocked data would make development much smoother. But not necessary for merging I think.

I'll do my best to review the changes today, but if not, I'll definitely take a look by tomorrow!

EllarBooher commented:
Thanks for taking a look! I'm definitely no expert in React so there's probably quite a few naive mistakes, but I'm doing my best to get better.

- I pushed a bunch of changes to get the layout working on mobile. I found the results quite good on my phone (~800 pixels wide) and while testing on my desktop (manually set to ~320 pixels). There are some minor issues at certain breakpoints, but all elements are visible and interactable as far as I can tell. 
- For testing, I was referring to the public feature of an example group with fake data, but a flexible endpoint would be nice too.
- I like your layout changes, thanks for fixing that stuff up.
- For the skill line chart, it had a bug where I was not handling responsiveness correctly. I believe I have fixed it with the most recent commit, and it should now take up all the horizontal main content space it can. Let me know if there's more you want to do, but maybe any further changes should occur with https://github.com/wouterrutgers/group-ironmen/issues/21.
- I'll take a look at the tool tip lag.
- I added `storage/cache` to the prettier ignore file. Are there any other auto-generated folders we need to ignore?

wouterrutgers commented on resources/components/tooltip/tooltip.css:
```diff
@@ -8,7 +8,7 @@
   transform: translate(5px, calc(-100% - 5px));
   min-width: 160px;
-  background-color: rgb(0 0 0;
+  background-color: rgb(0 0 0);
```
Oopsie! I did some cmd-z things and probably messed this up ;-)!

wouterrutgers commented:
> Thanks for taking a look! I'm definitely no expert in React so there's probably quite a few naive mistakes, but I'm doing my best to get better.
> 
> * I pushed a bunch of changes to get the layout working on mobile. I found the results quite good on my phone (~800 pixels wide) and while testing on my desktop (manually set to ~320 pixels). There are some minor issues at certain breakpoints, but all elements are visible and interactable as far as I can tell.

Mobile seems to work great now!

> * For testing, I was referring to the public feature of an example group with fake data, but a flexible endpoint would be nice too.

Yea, I figured out later when I opened the site and missed the demo button. I think it would be cool to have to back. Maybe if we have choose to host this instead of self hosted. When self hosted I don't feel that it's necessary to have a demo interface.

> * I like your layout changes, thanks for fixing that stuff up.

Awesome!

> * For the skill line chart, it had a bug where I was not handling responsiveness correctly. I believe I have fixed it with the most recent commit, and it should now take up all the horizontal main content space it can. Let me know if there's more you want to do, but maybe any further changes should occur with [Feature: More granular skill graph #21](https://github.com/wouterrutgers/group-ironmen/issues/21). EDIT: Upon further thought, by full page width do you mean hiding the sidebars? An easily accessible button that hides the sidebars could be good for this

I meant that the chart had was like max 70% of the container, on my screen anyway. A button that hides the sidebars could still be a cool thing!

And I think this is really compact compared to the other interfaces:

<img width="1205" height="265" alt="image" src="https://github.com/user-attachments/assets/a2bbd934-e333-4f1d-a7b1-18dc063c6232" />

But that's something I can look into later too!

> * I'll take a look at the tool tip lag.

It seems to be a lot better! :D

> * I added `storage/cache` to the prettier ignore file. Are there any other auto-generated folders we need to ignore?

Yea, some files used to generate all assets, I've added them to the ignore list in the PR mentioned above!

EllarBooher commented:
At this point, I think the changes are good to start iterating on. Feel free to take all the time you'd like to test. I left a comment on the cache busting PR, after that it should be ready to merge.

For some context on the lag issue, at least on my machine I traced it to how heavy the Items page renders are. I was getting ~170 ms for rendering all the item panels on a page since they are about 1 ms each. This was happening every single fetch to the backend since nearly every component on the page naively depended on the same context updating. The skill graph also has a similar issue that needs to be addressed, but it is not nearly as bad. The main ways I addressed this were:
- Use a reducer that sparingly updates the whole group's item dictionary, so that the items page only renders when items change, not every single fetch from the backend
- Use virtual scrolling to only update as many item panels as are visible, bringing down updates to like 20-50 ms depending on conditions/zoom. This has the downside of gaps where the elements are loading, but that could eventually be addressed with placeholders.

EllarBooher commented:
My bad, my rebase broke the verification of your authored commits. I hope that's not an issue, maybe I should have just done a normal merge.

wouterrutgers commented:
I'm not running it in production 🎉!!

Regarding the large JSON import with all the image hashes, I agree that it's a good idea to refactor it to use import() for asynchronous loading.

For minor tweaks or bug fixes, I think it's fine for either of us to commit directly to master. If we want feedback or are making more substantial changes, let's use pull requests to review each other's work.

Also, in case you miss my other comment because of my new comment;

> Btw, I assume you play right? What's your ingame name and group name (OetOetBtw - BringAKnife for me!)?

wouterrutgers commented:
Hehe, we've been playing for a while! Since september 2021 :)!

Awesome progress, I remember getting my first enhanced weapon seed, such dopamine!

I have you added in game btw :)!
